clappr/clappr-ios
# Fix Drawer Exit Fullscreen

## Goal

Fixes the drawer anchors when exiting fullscreen 

## How to Test

On `ViewController.swift` insert:

`Line #94`
```swift
Player.register(plugins: [CustomBottomDrawerPlugin.self])
```

`Line #110`
```swift
class CustomBottomDrawerPlugin: BottomDrawerPlugin {
    open class override var name: String {
        return "CustomBottomDrawerPlugin"
    }
    override var placeholder: CGFloat {
        return 32.0
    }
    override func render() {
        super.render()
        view.backgroundColor = .red
    }
}
```

1 - Portrait Mode

- Run the Clappr_Example and verify if the CustomDrawer is showing and interactable
- Click on Fullscreen button
- The drawer must be resized with the new screen size and interactable
- Click on Fullscreen button
- The drawer must be resized with the new screen size and interactable in the portrait mode

![Simulator Screen Shot - iPhone SE (2nd generation) - 2020-08-31 at 09 51 59](https://user-images.githubusercontent.com/28604291/91721820-bfccf900-eb6f-11ea-91b1-b3255240f087.png)

![Simulator Screen Shot - iPhone SE (2nd generation) - 2020-08-31 at 09 52 04](https://user-images.githubusercontent.com/28604291/91721849-cbb8bb00-eb6f-11ea-81cf-5db7b8d66d3f.png)

2 - Landscape Mode

- Run the Clappr_Example and verify if the CustomDrawer is showing and interactable
- Click on Fullscreen button
- The drawer must be resized with the new screen size and interactable
- Click on Fullscreen button
- The drawer must be resized with the new screen size and interactable in the landscape mode

![Simulator Screen Shot - iPhone SE (2nd generation) - 2020-08-31 at 09 54 10](https://user-images.githubusercontent.com/28604291/91721952-f571e200-eb6f-11ea-8a71-bd2913f60b0e.png)
![Simulator Screen Shot - iPhone SE (2nd generation) - 2020-08-31 at 09 54 14](https://user-images.githubusercontent.com/28604291/91721957-f73ba580-eb6f-11ea-9178-2abc973635bf.png)
